src/functional/ffront/foast_passes/type_deduction.py

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# GT4Py Project - GridTools Framework
+#
+# Copyright (c) 2014-2021, ETH Zurich
+# All rights reserved.
+#
+# This file is part of the GT4Py project and the GridTools framework.
+# GT4Py is free software: you can redistribute it and/or modify it under
+# the terms of the GNU General Public License as published by the
+# Free Software Foundation, either version 3 of the License, or any later
+# version. See the LICENSE.txt file at the top-level directory of this
+# distribution for a copy of the license or check <https://www.gnu.org/licenses/>.
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
+import warnings
+from typing import Optional
+
+import functional.ffront.field_operator_ast as foast
+from eve import NodeTranslator
+
+
+def is_complete_type(fo_type: foast.DataType) -> bool:
+    if fo_type is None:
+        return False
+    elif isinstance(fo_type, foast.DeferredSymbolType):
+        return False
+    elif isinstance(fo_type, foast.SymbolType):
+        return True
+    return False
```

Comment:
Any reason to not to use the new pattern matching syntax here? Additionally, this function looks a lot like a custom typeguard (https://www.python.org/dev/peps/pep-0647/) to me, would it make sense to explicitly define it like one?

Reply:
I was not aware that black had already fixed the problem with parsing `match`.

Reply:
it is not strictly a type guard (it excludes also DeferredSymbolType which is included in SymbolType), but for practical purposes we can think of it as one.

Reply:
> it is not strictly a type guard (it excludes also DeferredSymbolType which is included in SymbolType), but for practical purposes we can think of it as one.

Ok, thanks for the explanation, it make sense.